src/main/java/sirius/web/health/console/ConsoleController.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.web.health.console;
+
+import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Watch;
+import sirius.kernel.di.GlobalContext;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.console.Command;
+import sirius.web.controller.BasicController;
+import sirius.web.services.InternalService;
+import sirius.web.controller.Routed;
+import sirius.web.http.WebContext;
+import sirius.web.security.Permission;
+import sirius.web.services.JSONStructuredOutput;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+/**
+ * Provides the glue logic between the system console UI and the {@link Command commands}.
+ */
+@Register
+public class ConsoleController extends BasicController {
+
+    /**
+     * Describes the permission required to access the system console.
+     */
+    public static final String PERMISSION_SYSTEM_CONSOLE = "permission-system-console";
+
+    @Part
+    private GlobalContext globalContext;
+
+    /**
+     * Renders the UI for the system console.
+     *
+     * @param ctx the request being handled
+     */
+    @Routed("/system/console")
+    @Permission(PERMISSION_SYSTEM_CONSOLE)
+    public void console(WebContext ctx) {
+        ctx.respondWith().template("/templates/system/console.html.pasta");
+    }
+
+    @Routed("/system/console/api")
+    @Permission(PERMISSION_SYSTEM_CONSOLE)
+    @InternalService
+    public void consoleApi(WebContext webContext, JSONStructuredOutput out) throws Exception {
+        Watch w = Watch.start();
+
+        CommandParser parser = new CommandParser(webContext.require("command").asString());
+        if (Strings.isEmpty(parser.parseCommand())) {
```

Comment:
Parse once

Reply:
parse command is a noop the 2nd and 3rd call...